lib/shopify-cli/tasks/ensure_env.rb

```diff
@@ -3,27 +3,84 @@
 module ShopifyCli
   module Tasks
     class EnsureEnv < ShopifyCli::Task
-      def call(ctx)
+      def call(ctx, regenerate: false, required: [:api_key, :secret])
         @ctx = ctx
-        Resources::EnvFile.read(ctx.root)
-      rescue Errno::ENOENT
-        ask
+        env_data = begin
+          Resources::EnvFile.parse_external_env
+                   rescue Errno::ENOENT
+                     {}
+        end
+
+        return {} if !regenerate && required.all? { |property| env_data[property] }
+
+        org = fetch_org
+        write_env(env_data, org)
+        org
+      end
+
+      private
+
+      def fetch_org
+        orgs = PartnersAPI::Organizations.fetch_with_app(@ctx)
+        org_id = if orgs.count == 1
+          orgs.first["id"]
+        else
+          CLI::UI::Prompt.ask(@ctx.message('core.tasks.ensure_env.organization_select')) do |handler|
+            orgs.each do |org|
+              handler.option(
+                ctx.message('core.partners_api.org_name_and_id', org['businessName'], org['id'])
```

Comment:
Does this work? I think we need to use the instance var here, right? `@ctx`? 🤔 
```suggestion
                @ctx.message('core.partners_api.org_name_and_id', org['businessName'], org['id'])
```